README.tfdoc.hcl

```diff
@@ -0,0 +1,1265 @@
+          `TCP_UDP`. Not valid to use `UDP` or `TCP_UDP` if dual-stack mode is
+          enabled. Not valid for Gateway Load Balancers.
+        END
+      }
+
+      variable "certificate_arn" {
+        type        = string
+        description = <<-END
+          ARN of the default SSL server certificate. Exactly one certificate is
+          required if the protocol is `HTTPS`.
+        END
+      }
+
+      variable "alpn_policy" {
+        type        = string
+        default     = "None"
+        description = <<-END
+          Name of the Application-Layer Protocol Negotiation (ALPN) policy. Can
+          be set if `protocol` is set to `TLS`. Valid values are `HTTP1Only`,
+          `HTTP2Only`, `HTTP2Optional`, `HTTP2Preferred`, and `None`.
+        END
+      }
+
+      variable "ssl_policy" {
+        type        = string
+        default     = "ELBSecurityPolicy-2016-08"
+        description = <<-END
+          Name of the SSL Policy for the listener. Required if protocol is
+          `HTTPS` or `TLS`.
+        END
+      }
+
+      variable "additional_certificates_arns" {
+        type        = set(string)
+        default     = []
+        description = <<-END
+          The ARNs of the additional certificates to attach to the listener.
+        END
+      }
+
+      variable "default_action" {
+        type = object(default_action)
+        default = {
+          fixed_response = {
+            content_type = "plain/text"
+            message_body = "Nothing to see here!"
+            status_code  = 418
+          }
+        }
+        description = <<-END
+          Configuration block for default actions.
+        END
+
+        attribute "type" {
+          required    = true
+          type        = string
+          description = <<-END
+            Type of routing action. Valid values are `forward`, `redirect`,
+            `fixed-response`, `authenticate-cognito` and `authenticate-oidc`.
+          END
+        }
+
+        attribute "target_group_arn" {
+          type        = string
+          description = <<-END
+            ARN of the Target Group to which to route traffic. Specify only if
+            `type` is `forward` and you want to route to a single target group.
+            To route to one or more target groups, use a `forward` block instead.
+          END
+        }
+
+        attribute "order" {
+          type        = number
+          description = <<-END
+            Order for the action. This value is required for rules with multiple
+            actions. The action with the lowest value for order is performed
+            first. Valid values are between `1` and `50000`.
+          END
+        }
+
+        attribute "authenticate_cognito" {
+          type        = object(authenticate_cognito)
+          default     = []
+          description = <<-END
+            Configuration block for using Amazon Cognito to authenticate users.
+            Specify only when `type` is `authenticate-cognito`.
```

Comment:
```suggestion
```

This is not required as we abstract away the complexity (type isn't required in this case)

Reply:
for example: https://github.com/mineiros-io/terraform-aws-lb-listener/blob/main/main.tf#L41